MetaMask/metamask-extension
# Feat/add snap account label

## **Description**

This feature adds a label with the derived snap name and MetaMask snap icon to the tag on the accounts list and wallet overview UI.

Reason: Account snaps should present a snap tag, including the snap badge and the name of the snap it's derived from.

Improvement: The account-list-menu has been updated to pass a keyring type and optional tag label to child account-list-item components. The wallet-overview screen now shows a tag with the snap name for snap accounts.

## **Related issues**

Fixes: #20819 

## **Manual testing steps**

1. Go to the [SSK](https://metamask.github.io/snap-simple-keyring/latest/) and create a snap account
2. Open the accounts list dropdown in the extension
3. You should see "MetaMask Simple Snap Keyring (Beta)" under the new account with the snap icon
4. Open the expanded view of the extension
5. You should see "MetaMask Simple Snap Keyring (Beta)" under the new account with the snap icon when you navigate to the new account page 
6. You should NOT see any tag under an account that is not a snap account (default MM accounts)

### **Jest**
`jest ui/components/multichain/account-list-menu/account-list-menu.test.js`
`jest ui/components/multichain/account-list-item/account-list-item.test.js`

## **Screenshots/Recordings**

### **Before**

<img width="320" alt="Screenshot 2023-12-14 at 2 25 55 PM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/6e7a0726-f87f-412d-9ee8-f8818e1768a7">

<img width="968" alt="Screenshot 2023-12-14 at 2 26 34 PM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/88e196eb-0348-4fa6-8506-f53e1f555a6f">



### **After**

<img width="330" alt="Screenshot 2023-12-14 at 2 29 24 PM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/e5c53c61-8dfb-4fce-8da5-0427d7ba3a52">

<img width="988" alt="Screenshot 2023-12-14 at 2 30 04 PM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/6df8e849-7647-49fc-9f59-5f9bfdceb9f9">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
